backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/snapshots/RemoveSnapshotSingleDiskLiveCommand.java

```diff
@@ -182,6 +183,12 @@ private RemoveSnapshotSingleDiskStep getInitialMergeStepForImage(Guid imageId) {
                 return RemoveSnapshotSingleDiskStep.DESTROY_IMAGE;
             }
         }
+        if (getImageInfoFromVdsm(getDestinationDiskImage()) == null) {
+            log.info("Image has been previously merged and deleted, proceeding to DB sync");
+            getParameters().setMergeStatusReturnValue(
+                    new MergeStatusReturnValue(Collections.singleton(getDestinationDiskImage().getImageId())));
```

Comment:
This `MergeStatusReturnValue` is used only here, right? No deserialization is / might be involved, right?
As I learnt from Benny and by introducing a bug using structures like `Collections.singletonMap()`, `Collections.emptyMap()` or even double-brace initialization (that creates an inner class) in parameters is a bad idea :)
https://github.com/oVirt/ovirt-engine/pull/305#discussion_r895634710
https://github.com/oVirt/ovirt-engine/pull/526

Reply:
yeah, whatever we store in the parameters needs to be deserializable (since the engine might or the cache might be cleared during the execution of the command)